File: internal/sem/testdata/fixtures/csharp-oo/Shapes.cs
using System;

namespace Shapes
{
    public interface IShape
    {
        double Area();
    }

    public abstract class Base
    {
        public abstract string Describe();
    }

    public class Circle : Base, IShape
    {
        public double Area()
        {
            return 3.14;
        }

        public override string Describe()
        {
            return "circle";
        }
    }
}

File: internal/sem/testdata/fixtures/php-oo/Shapes.php
<?php

namespace Shapes;

interface Shape
{
    public function area(): float;
}

abstract class Base
{
    abstract public function describe(): string;
}

class Circle extends Base implements Shape
{
    public function area(): float
    {
        return 3.14;
    }

    public function describe(): string
    {
        return "circle";
    }
}
